### ISSUE TITLE
Source map doesn't work

### ISSUE THREAD

**Issue by dfedynich** [react-cosmos/react-cosmos#346]
Hello, guys. I feel pleasure while using cosmos to debug my React components like in real sand-box. Great DX tool indeed!

However I found one issue, which decreases the speed of my debugging session a lot. I use devtool **'source-map'** in my webpack.config.js and it seems that everything goes as expected.

![image](https://cloud.githubusercontent.com/assets/7403902/25530756/11680c26-2c30-11e7-9ae0-cb8e1aab8ce6.png)

However when I mark a debug point in my component's source code individual file, it doesn't fire. Could you give me any tips, why the source map doesn't work?

**Comment by skidding** [comment]
Did you by any chance check out [this thread](https://github.com/webpack/webpack/issues/2145#issuecomment-294361203)? Source maps with Webpack can be problematic, but it seems like `cheap-module-source-map` & `cheap-module-inline-source-map` are better alternatives so maybe you should give them a try. Please report your findings if you do. Thanks!

**Comment by dfedynich** [comment]
Hey, @skidding, thank you for the link with comparison of devtools for webpack. If we combine these knowledge with [webpack official](https://webpack.github.io/docs/configuration.html#devtool), we will get a real combo. However, I didn't manage to achieve a success - still can't fire breakpoints.

Here is my relevant webpack.config.js file
```
'use strict';

var merge = require('webpack-merge');
var devConfig = require('./webpack.dev');
var HtmlWebpackPlugin = require('html-webpack-plugin');

var config = merge(devConfig, {
    devtool: '#cheap-module-source-map',
    plugins: [
        new HtmlWebpackPlugin({
            filename: './index.html'
        })
    ]
});

module.exports = config;
```

Here is my cosmos.config.js file
```
module.exports = {
    componentPaths: [
           //my path to the components
    ],

    getComponentName: function (componentPath) {
        return componentPath.match(/components\/(.+)\.js$/)[1]
    },

    globalImports: [
        './vendor',
        './main'
    ],

    proxies: [
        'react-cosmos-normalize-props-proxy',
        './redux-proxy'
    ],

    hot: true,
    hmrPlugin: true,

    hostname: 'localhost',
    port: 8989,
    webpackConfigPath: './webpack.config.js'
};
```

Do you have any thoughts, why breakpoints are not fired?

**Comment by david-ogalego** [comment]
I am having the same problem. Is it possible that as the bundle generated, is being loaded twice, the browser does not know how to link the sourcemap?

**Comment by david-ogalego** [comment]
We make some changes in our fork to make some changes in react-cosmos-webpack.
https://github.com/BeezyInc/react-cosmos/commit/4b99394ecf5e4a3db490bf87d000aaf2ef3a0f83
 
And in our custom webpack.config configuration inject one of this bundles:

`new HtmlWebpackPlugin({
	chunks: ['preview'],
	inject: true,
	template: paths.getConfigPaths().appHtml,
}),`

It is working in our case.

**Comment by dfedynich** [comment]
Hello, @skidding, do you have any thoughts about this issue?
Could you include the changes from @david-ogalego fork to resolve this issue?

**Comment by skidding** [comment]
Hi @dfedynich, I'm currently away at [JSHeroes](http://jsheroes.io/) (gave a talk yesterday and got more people interested in React Cosmos :D). I'll take a closer look next week to understand the problem and I'd be happy to include a fix for this.

**Comment by dfedynich** [comment]
Hey, @david-ogalego, you was right. **bundle.js** is loading twice. Here is the network from Chrome inspector.

![image](https://user-images.githubusercontent.com/7403902/26981711-a53a18e8-4d3e-11e7-98d9-d6300dce70b4.png)

**Comment by skidding** [comment]
Hmm. When #364 is merged the bundles (playground & loader) will be different and maybe it changes the situation.

**Comment by dfedynich** [comment]
Hey, team. Could you reveal you expectations when the merging process of #364 will be done?

**Comment by skidding** [comment]
I'll ask for code review some time next week. Thanks for the patience! 🤞

**Comment by skidding** [comment]
Hey @dfedynich. #364 has been merged and released under `react-cosmos-webpack@2.0.0-beta.16`. Please let me know if the source code issue is still present in the last version. Thanks!
